app/src/main/java/com/genkit/restaurant/domain/viewmodel/AuthViewModel.kt

```
@@ -0,0 +1,181 @@
+                    if (sessionData != null) {
+                        _uiState.value = AuthUiState.Success(sessionData)
+                    } else {
+                        _uiState.value = AuthUiState.Error("Failed to create session")
+                    }
+                } else {
+                    val exception = result.exceptionOrNull()
+                    _uiState.value = AuthUiState.Error(
+                        exception?.message ?: "Something went wrong, please try again"
+                    )
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)